用來處理Error Handing

在使用do try catch時，我們必須將呼叫可能會拋出錯誤的方法及之後要做的事情包在 do{} 裡面，並在呼叫該方法前擺上 try 關鍵字，其後將錯誤出現時要做的事情放在 catch{}裡。

在 do try catch 中擁有catch的功能，所以可以理所當然的接到一個 error

什麼時候才要Error Handling?

這些 api 都擁有這個關鍵字 throws ，代表使用這些方法的時候要特別小心，因為一個使用不慎，這些方法會 throws （ 拋 ）出錯誤來。

前文有提到，當這些方法拋出錯誤時，我們可以適時的印出這些錯誤來讓開發者知道錯誤是如何發生的，而這些錯誤是由寫這些 api 的開發者所訂定。

當然，我們也可以自訂一個會拋出錯誤訊息的 funct

try try? try!的差別?

try
當開發者需要很完整的錯誤訊息，或是依據每個所拋出的錯誤的差異而有不同的事情要做時

try?

由於 try? 不擁有 catch 功能，所以當方法拋出錯誤訊息時，開發者僅能得知有錯誤發生，卻不會得知錯誤訊息的種類。


try!

最簡潔的 Error Handling 就屬 try! 莫屬。但伴隨著執行錯誤的風險。所以在使用時要特別小心，會使用 try! 通常都是開發人員非常非常確定不會有錯誤產生的時候
try try? try!的使用時機?




